File: arrange.py
#import sys
#i = 0
#i = i+1
    #print(i)
    #if i > 10:
    #    sys.exit()

def arrange(remain, build):
    
    if (len(build) == 3):
        print(build)
    else:
        for elements in remain:
            buildCopy = build + remain[-1:]
            remainCopy = remain[:-1]
            arrange(remainCopy, buildCopy)




def arrange2(remain, build):
    if len(build) == 3:
        print(build)
    else:
        for i in range(len(remain)):
            new_build = build + [remain[i]]
            new_remain = remain[:i] + remain[i+1:]
            arrange(new_remain, new_build)
    

arrange2([1, 2, 3], [])

File: arrange.rs
fn arrange(remain: &mut Vec<i32>, build: &mut Vec<i32>) {
    if remain.is_empty() {
        println!("{:?}", build);
    } else {
        for i in 0..remain.len() {
            let mut build_copy = build.clone();
            build_copy.push(remain[i]);
            let mut remain_copy = remain.clone();
            remain_copy.remove(i);
            arrange(&mut remain_copy, &mut build_copy);
        }
    }
}

fn main() {
    let mut data = vec![1, 2, 3, 4];
    let mut result = vec![];
    arrange(&mut data, &mut result);
}



fn arrange(remain: &mut Vec<i32>, build: &mut Vec<i32>) {
    if remain.is_empty() {
        println!("{:?}", build);
    } else {
        for i in 0..remain.len() {
            let mut build_copy = build.clone();
            let last = remain[i];
            build_copy.push(last);
            let mut remain_copy = remain.clone();
            remain_copy.remove(i);
            arrange(&remain_copy, build_copy);
        }
    }
}

fn main() {
    let mut data = vec![1, 2, 3, 4];
    let mut result = vec![];
    arrange(&mut data, &mut result);
  
}
